File: app/src/main/java/com/amarnehsoft/holyquran/db/DBModule.java
package com.amarnehsoft.holyquran.db;

import android.arch.persistence.room.Room;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DBModule {
    @Provides
    @Singleton
    public AppDatabase provideDB(Context context){
        return Room.databaseBuilder(context,
                AppDatabase.class, "holy-quran").allowMainThreadQueries().build();
    }

    @Provides
    @Singleton
    public SuraDao provideSurahDao(AppDatabase database){
        return database.suraDao();
    }

    @Provides
    @Singleton
    public AyaDao provideAyaDoa(AppDatabase database){
        return database.ayaDao();
    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/db/DBModule.kt
package com.amarnehsoft.holyquran.db

import androidx.room.Room
import android.content.Context
import androidx.room.RoomDatabase

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class DBModule {

    @Provides
    @Singleton
    fun provideRoomDatabase(appDatabase: AppDatabase): RoomDatabase {
        return appDatabase
    }

    @Provides
    @Singleton
    fun provideAppDatabase(context: Context): AppDatabase {
        return Room.databaseBuilder(
            context,
            AppDatabase::class.java, "holy-quran"
        ).fallbackToDestructiveMigration().build()
    }

    @Provides
    @Singleton
    fun provideSurahDao(database: AppDatabase): SuraDao {
        return database.suraDao()
    }

    @Provides
    @Singleton
    fun provideAyaDoa(database: AppDatabase): AyaDao {
        return database.ayaDao()
    }
}
